### PR TITLE
Add Cask for BitPim.App  (CDMA phone explorer) v1.0.7

### PR DESCRIPTION
**Pull request by johntrandall**
This is my first commit to an open-source project, so please be kind. Constrictive criticism would be appreciated (please ELI5).

Not sure if I did the 'uninstall' and 'zap' stanzas correctly...  I figure that the 'quit' is needed for zap because you don't want those pref files being re-written if the app is still open. But does the 'quit' belong in the 'uninstall' stanza as well? 

FYI - Is one of the best GUI tools for pulling data from old phones. It hasn't been updated in a few years, but it still works on 10.9 (as tested by me), and probably 10.10.

### CODE REVIEW

File: Casks/bitpim.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+cask :v1 => 'bitpim' do
+  version '1.0.7'
+  sha256 '567d41ececc8c416746c3aa9365182797b339001d255dc4da7acc285bb289880'
+  url 'http://downloads.sourceforge.net/project/bitpim/bitpim/1.0.7/LEOPARD-bitpim-1.0.7.dmg'
```
Comment (jawshooah) [inline Casks/bitpim.rb]: Version interpolation is preferred:
```ruby
url "http://downloads.sourceforge.net/project/bitpim/bitpim/#{version}/LEOPARD-bitpim-#{version}.dmg"
```

File: Casks/bitpim.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+  name 'BitPim'
+  homepage 'http://www.bitpim.org/'
+  license :gpl
+  app 'BitPim.app'
+  depends_on :macos => '>= 10.3'
+
+  uninstall :quit => 'org.bitpim.bitpim'
+
+  zap :quit => 'org.bitpim.bitpim'
```
Comment (jawshooah) [inline Casks/bitpim.rb]: `brew cask zap` will first run the uninstaller, so this line is not needed.

File: Casks/bitpim.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  zap :delete => [
+    '~/Library/Preferences/org.bitpim.bitpim.plist',
+    '~/Library/Preferences/LaunchpadCleaner/Local\ Store/applicattons/appicons/org.bitpim.bitpim.png',
```
Comment (jawshooah) [inline Casks/bitpim.rb]: You don't need to escape spaces here.

**Comment by vitorgalvao** [comment]
Thank you for the contribution. Some refinements are needed, yes. I’ll comment on each line and then write how the cask should be in the end (so you can copy and paste), mainly because the order of stanzas is wrong (functionally it works, but is inconsistent with other casks).

```ruby
cask :v1 => 'bitpim' do
  version '1.0.7'
  sha256 '567d41ececc8c416746c3aa9365182797b339001d255dc4da7acc285bb289880'

  url "http://downloads.sourceforge.net/project/bitpim/bitpim/#{version}/LEOPARD-bitpim-#{version}.dmg"
  name 'BitPim'
  homepage 'http://www.bitpim.org/'
  license :gpl

  app 'BitPim.app'

  uninstall :quit => 'org.bitpim.bitpim'

  zap :delete => [
    '~/Library/Preferences/org.bitpim.bitpim.plist',
    '~/Library/Preferences/LaunchpadCleaner/Local Store/applicattons/appicons/org.bitpim.bitpim.png',
  ]

  depends_on :macos => '>= :tiger'
end
```

File: Casks/bitpim.rb [hunk 4]
```diff
@@ -0,0 +1,19 @@
+cask :v1 => 'bitpim' do
+  version '1.0.7'
+  sha256 '567d41ececc8c416746c3aa9365182797b339001d255dc4da7acc285bb289880'
+  url 'http://downloads.sourceforge.net/project/bitpim/bitpim/1.0.7/LEOPARD-bitpim-1.0.7.dmg'
+
+  name 'BitPim'
+  homepage 'http://www.bitpim.org/'
+  license :gpl
+  app 'BitPim.app'
+  depends_on :macos => '>= 10.3'
```
Comment (vitorgalvao) [inline Casks/bitpim.rb]: [There’s no `10.3`](https://github.com/caskroom/homebrew-cask/blob/master/doc/CASK_LANGUAGE_REFERENCE.md#depends_on-macos). Also, we prefer to call them out by release name.

File: Casks/bitpim.rb (hunk 3 above)
Comment (jawshooah) [inline Casks/bitpim.rb]: Actually, this line doesn't belong at all. This file is a component of `LaunchpadCleaner`, not `BitPim`, and thus does not belong in this cask.

File: Casks/bitpim.rb (hunk 1 above)
Comment (vitorgalvao) [inline Casks/bitpim.rb]: The changes I’ve made here are optional, they just make [updates easier](https://github.com/caskroom/homebrew-cask/pull/4910).

File: Casks/bitpim.rb (hunk 2 above)
Comment (vitorgalvao) [inline Casks/bitpim.rb]: `zap` always runs `uninstall` first, so this is unnecessary.

File: Casks/bitpim.rb (hunk 3 above)
Comment (vitorgalvao) [inline Casks/bitpim.rb]: The `\` is wrong here, because the string is quoted, meaning the slash would literally be there. Also, what is this you’re removing? It seems to not belong to the app in this cask.

File: Casks/bitpim.rb [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+  uninstall :quit => 'org.bitpim.bitpim'
```
Comment (johntrandall) [inline Casks/bitpim.rb]: @vitorgalvao You mentioned that 'zap' calls the uninstall stanza first. But is the 'quit' needed, or for that matter, welcome, in the uninstall Stanza?

Comment (vitorgalvao) [inline Casks/bitpim.rb]: It isn’t needed, no, but it also isn’t harmful. Few casks do it in this situation, and [we should eventually start doing it automatically.](https://github.com/caskroom/homebrew-cask/pull/7801).

You can remove it, yes. Nice catch. After making changes, please always [squash your commits](http://davidwalsh.name/squash-commits-git).
